dlrover/python/unified/controller/manager.py

```diff
@@ -219,12 +357,36 @@ async def wait(self):
         await self._stopped_event.wait()
         assert self.stage == MasterStage.STOPPED
 
+    def _get_state_key(self):
+        return MasterConstant.MASTER_STATE_KEY_PREFIX + self.config.job_name
+
     def save(self):
         """Save the job state to persistent storage."""
         # This is a placeholder for saving the job state.
         # In a real implementation, this would save to a database or file.
-        logger.info(
-            f"Job state saved: {self.stage}, "
-            f"nodes: {[node.name for node in self.graph.vertices]}"
-        )
-        # TODO implement actual save logic. (Ref state_backend.py)
+
+        state_key = self._get_state_key()
+        self._state_backend.set(state_key, self._context.serialize())
+
+        logger.info("Save runtime context into state.")
+
+    def _init_or_reload(self) -> Tuple[bool, ManagerRuntimeContext]:
```

Comment:
May be it's better keep this function simpler, only `try_load`, keep init in `__init__`.

Reply:
need to ````try_load``` 1st , than ````try_init``` if so, still need 2 conditions check to ensure 'init' will not override the 'reload' state